packages/bee-q/src/components/breadcrumb/bq-breadcrumb.tsx

```diff
@@ -0,0 +1,135 @@
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Handler to be called when `bq-breadcrumb-item` item loses focus. */
+  @Event() bqBreadcrumbBlur: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when `bq-breadcrumb-item` item gets focus. */
+  @Event() bqBreadcrumbFocus: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  /** Handler to be called when `bq-breadcrumb-item` is selected (on click/enter press). */
+  @Event() bqBreadcrumbClick: EventEmitter<HTMLBqBreadcrumbItemElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('bqBlur', { passive: true })
+  onBlur(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbBlur.emit(event.detail);
+  }
+
+  @Listen('bqFocus', { passive: true })
+  onFocus(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbFocus.emit(event.detail);
+  }
+
+  @Listen('bqClick', { passive: true })
+  onClick(event: CustomEvent<HTMLElement>) {
+    if (isHTMLElement(event.detail, 'bq-breadcrumb-item')) this.bqBreadcrumbClick.emit(event.detail);
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private setSeparator = (): void => {
+    this.breadcrumbItems.forEach((item, index, arr) => {
+      item.isLastItem = index === arr.length - 1;
+      !item.isLastItem && item.append(this.getSeparatorElem());
+    });
+  };
+
+  /**
+   * clone original element and add slot attr
+   * @returns cloned separator element
+   */
+  private getSeparatorElem = (): HTMLElement => {
+    const separator = this.separatorFromSlot || this.defaultSeparator;
+    const clone = separator.cloneNode(true) as HTMLElement;
+    clone.slot = 'separator';
+
+    return clone;
+  };
+
+  private get breadcrumbItems(): HTMLBqBreadcrumbItemElement[] {
+    return Array.from(this.el.querySelectorAll('bq-breadcrumb-item'));
+  }
+
+  private get separatorFromSlot() {
+    return this.spanElem
+      .querySelector<HTMLSlotElement>('slot[name="separator"]')
+      .assignedElements({ flatten: true })[0] as HTMLElement;
+  }
+
+  private get defaultSeparator(): HTMLElement {
+    const newNode = document.createElement('span');
+    newNode.classList.add('flex', 'w-3', 'items-center', 'justify-center');
+    newNode.textContent = '/';
+
+    return newNode;
+  }
```

Comment:
You can add a `<bq-icon>` inside the slot separator and then the logic to clone it as you do for a custom one. That way you don't need to worry if a custom separator is set, to remove the default, it will be overridden by the external slot content.

Reply:
Nice. Will change it!